mux.go

```diff
@@ -59,6 +59,12 @@ func (r *Router) Match(req *http.Request, match *RouteMatch) bool {
 			return true
 		}
 	}
+
+	// Closest match for a router (includes sub-routers)
+	if r.NotFoundHandler != nil {
+		match.Handler = r.NotFoundHandler
+		return true
```

Comment:
If we don't report it as a match we would need to re-engineer how sub-router 404 handling works (e.g. revert this).

The alternative would be to set `match.Handler == nil` and then add another field to `Router` to handle this behaviour.